src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -338,66 +333,120 @@ func KafkaAuthConfig(
 				sarama.SASLTypeSCRAMSHA512,
 			)
 		}
-		return &config.SCRAM{
-			User:     u,
-			Password: p,
-			Type:     m,
+		return &config.SASL{
+			User:      u,
+			Password:  p,
+			Mechanism: m,
 		}, nil
 	}
 }
 
-func BuildTLSConfig(
-	certFile,
-	keyFile,
-	truststoreFile *string,
+func BuildAdminApiTLSConfig(
+	certFile, keyFile, truststoreFile *string,
+	configuration func() (*config.Config, error),
 ) func() (*config.TLS, error) {
 	return func() (*config.TLS, error) {
-		c := *certFile
-		k := *keyFile
-		t := *truststoreFile
+		defaultVal := func() (*config.TLS, error) {
+			conf, err := configuration()
+			if err != nil {
+				return nil, err
+			}
+			return conf.Rpk.AdminApi.TLS, nil
+		}
+		return buildTLS(
+			certFile,
+			keyFile,
+			truststoreFile,
+			"REDPANDA_ADMIN_TLS_CERT",
+			"REDPANDA_ADMIN_TLS_KEY",
+			"REDPANDA_ADMIN_TLS_TRUSTSTORE",
+			defaultVal,
+		)
+	}
+}
 
-		if c == "" {
-			c = os.Getenv("REDPANDA_TLS_CERT")
-		}
-		if k == "" {
-			k = os.Getenv("REDPANDA_TLS_KEY")
-		}
-		if t == "" {
-			t = os.Getenv("REDPANDA_TLS_TRUSTSTORE")
-		}
+func BuildKafkaTLSConfig(
+	certFile, keyFile, truststoreFile *string,
+	configuration func() (*config.Config, error),
+) func() (*config.TLS, error) {
+	return func() (*config.TLS, error) {
+		defaultVal := func() (*config.TLS, error) {
+			conf, err := configuration()
+			if err != nil {
+				return nil, err
+			}
+			return conf.Rpk.KafkaApi.TLS, nil
+		}
+		return buildTLS(
+			certFile,
+			keyFile,
+			truststoreFile,
+			"REDPANDA_TLS_CERT",
+			"REDPANDA_TLS_KEY",
+			"REDPANDA_TLS_TRUSTSTORE",
+			defaultVal,
+		)
+	}
+}
 
-		if t == "" && c == "" && k == "" {
-			return nil, nil
-		}
-		if t == "" && (c != "" || k != "") {
-			return nil, fmt.Errorf(
-				"--%s is required to enable TLS",
-				truststoreFileFlag,
-			)
-		}
-		if c != "" && k == "" {
-			return nil, fmt.Errorf(
-				"if --%s is passed, then --%s must be passed to enable"+
-					" TLS authentication",
-				certFileFlag,
-				keyFileFlag,
-			)
-		}
-		if k != "" && c == "" {
-			return nil, fmt.Errorf(
-				"if --%s is passed, then --%s must be passed to enable"+
-					" TLS authentication",
-				keyFileFlag,
-				certFileFlag,
-			)
-		}
-		tls := &config.TLS{
-			KeyFile:        k,
-			CertFile:       c,
-			TruststoreFile: t,
-		}
-		return tls, nil
+// Builds an instance of config.TLS.
+// If certFile, keyFile or truststoreFile are nil, then their corresponding
+// env vars are checked (given by certEnvVar, keyEnvVar & truststoreEnvVar).
+// If after that no value is found for any of them, the result of calling
+// defaultVal is returned.
+func buildTLS(
```

Comment:
at 9 arguments, I think it starts to become visually easier to read if the arguments are moved to a dedicated struct type

Reply:
I wish Go had named args. What would you name the struct?

Reply:
If I'm being uncreative, `buildTlsParams` or `buildTLSParams`.

Reply:
Hehe sounds good.

Reply:
On second thought, I think this creates an unnecessary point of indirection. It's 7 args, which isn't as few as one would usually like, but it's not 9 hehe.